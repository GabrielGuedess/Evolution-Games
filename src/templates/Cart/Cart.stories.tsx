import { Story, Meta } from '@storybook/react/types-6-0';
import {
  CartContext,
  CartContextDefaultValues,
  CartContextProps,
} from 'hooks/useCart/useCart';

import { Cart } from './Cart';
import items from './mock';

export default {
  title: 'Templates/Cart',
  component: Cart,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Empty: Story = () => <Cart />;

export const Populated: Story = () => {
  const cartProviderProps: CartContextProps = {
    ...CartContextDefaultValues,
    items,
    total: 'R$300,00',
  };

  return (
    <CartContext.Provider value={cartProviderProps}>
      <Cart />
    </CartContext.Provider>
  );
};
