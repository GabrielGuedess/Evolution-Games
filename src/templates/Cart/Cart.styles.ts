import media from 'styled-media-query';

import { Container } from 'components/atoms/Container/Container';

import styled, { css } from 'styled-components';

export const Wrapper = styled(Container)`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: 4.6rem;
    margin-top: ${theme.spacings.large};

    ${media.greaterThan('large')`
      flex-direction: row;
      min-height: 70vh;
    `}
  `}
`;

export const Heading = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding-bottom: ${theme.spacings.small};
    border-bottom: 1px solid ${theme.colors.whiteText};
    margin-bottom: ${theme.spacings.xsmall};

    ${media.greaterThan('medium')`
      padding-bottom: ${theme.spacings.large};
    `}
  `}
`;

export const FirstColumn = styled.div`
  width: 100%;

  ${media.greaterThan('large')`
    width: 70%;
  `}
`;

export const SecondColumn = styled.div`
  width: 100%;
  height: auto;
  position: relative;

  ${media.greaterThan('large')`
    width: 30%;
  `}
`;

export const Summary = styled.div`
  ${({ theme }) => css`
    min-height: 40rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    background-color: ${theme.colors.gameDetails};
    border-radius: ${theme.border.radius.regular};
    padding: ${theme.spacings.large};
    position: sticky;
    top: 0;
  `}
`;

export const SummaryHeader = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
`;

export const Total = styled.strong`
  ${({ theme }) => css`
    font-size: 1.6rem;
    color: ${theme.colors.whiteText};
  `}
`;

export const EmptyBox = styled.div`
  width: 100%;
  height: 70vh;
  margin-top: 9rem;
`;
