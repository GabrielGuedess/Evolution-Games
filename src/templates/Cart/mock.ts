import { cartItems } from 'components/organisms/Navbar/mock';

export default cartItems;
