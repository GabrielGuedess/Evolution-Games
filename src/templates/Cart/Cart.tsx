import { useCart } from 'hooks/useCart/useCart';

import { Base } from 'templates/Base/Base';

import Button from 'components/atoms/Button/Button';
import { Empty } from 'components/atoms/Empty/Empty';
import { Title } from 'components/atoms/Title/Title';
import { GameCartDrop } from 'components/molecules/GameCartDrop/GameCartDrop';
import { HighlightCarousel } from 'components/organisms/HighlightCarousel/HighlightCarousel';
import { highlightMock } from 'components/organisms/HighlightCarousel/mock';

import * as S from './Cart.styles';

export const Cart = () => {
  const { items, total } = useCart();

  return (
    <Base hasColors hasMarginTop>
      <S.Wrapper>
        {items.length > 0 ? (
          <>
            <S.FirstColumn>
              <S.Heading>
                <Title textAlign="left">Carrinho</Title>
              </S.Heading>
              {items.map(game => (
                <GameCartDrop key={game.title} {...game} />
              ))}
            </S.FirstColumn>
            <S.SecondColumn>
              <S.Summary>
                <S.SummaryHeader>
                  <S.Total>Valor dos produtos:</S.Total>
                  <S.Total>{total}</S.Total>
                </S.SummaryHeader>
                <Button>Ir para o pagamento</Button>
              </S.Summary>
            </S.SecondColumn>
          </>
        ) : (
          <S.EmptyBox>
            <Empty
              title="Nenhum jogo encontrado"
              description="Você não adicionou nenhum jogo a sua bolsa de compras."
            />
          </S.EmptyBox>
        )}
      </S.Wrapper>
      <HighlightCarousel title="Novidades" data={highlightMock} />
    </Base>
  );
};
