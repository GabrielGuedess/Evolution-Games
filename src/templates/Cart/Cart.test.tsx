import { renderWithProviders } from 'utils/tests/helpers';

import {
  CartContextProps,
  CartContextDefaultValues,
} from 'hooks/useCart/useCart';

import { screen } from '@testing-library/react';

import { Cart } from './Cart';
import items from './mock';

jest.mock('templates/Base/Base', () => ({
  __esModule: true,
  Base: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Base Mock">{children}</div>;
  },
}));

jest.mock('components/organisms/HighlightCarousel/HighlightCarousel', () => ({
  __esModule: true,
  HighlightCarousel: function Mock({
    children,
  }: {
    children: React.ReactNode;
  }) {
    return <div data-testid="HighlightCarousel">{children}</div>;
  },
}));

describe('<Cart />', () => {
  it('should render <Cart /> without any items', () => {
    // Arrange
    const { container } = renderWithProviders(<Cart />);

    // Assert
    expect(screen.getByText('Nenhum jogo encontrado')).toBeInTheDocument();
    expect(
      screen.getByText(
        'Você não adicionou nenhum jogo a sua bolsa de compras.',
      ),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render <Cart /> without any items', () => {
    // Arrange
    const cartProviderProps: CartContextProps = {
      ...CartContextDefaultValues,
      items,
    };

    renderWithProviders(<Cart />, { cartProviderProps });

    // Assert
    expect(screen.getByText('GTA 6')).toBeInTheDocument();
    expect(screen.getByText('The Witcher 3')).toBeInTheDocument();
    expect(screen.getByText('Rainbow Six')).toBeInTheDocument();
  });
});
