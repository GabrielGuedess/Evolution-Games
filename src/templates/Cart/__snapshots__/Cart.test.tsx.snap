// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Cart /> should render <Cart /> without any items 1'] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 6rem;
}

.c4 {
  max-width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3rem;
}

.c6 {
  font-weight: 700;
  font-size: 2.8rem;
  line-height: 34px;
  text-align: center;
  color: #855aff;
  text-shadow: 0px 0.8rem 2.4rem rgba(118,74,241,0.25);
}

.c7 {
  font-size: 1.8rem;
  text-align: center;
  color: #717B8A;
}

.c0 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4.6rem;
  margin-top: 4.0rem;
}

.c2 {
  width: 100%;
  height: 70vh;
  margin-top: 9rem;
}

@media (min-width:1170px) {
  .c0 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c0 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c0 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width:1170px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    min-height: 70vh;
  }
}

<div
  data-testid="Base Mock"
>
  <div
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <img
          alt="two gamers in a couch playing videogame"
          class="c4"
          src="/img/svg/Empty.svg"
        />
        <div
          class="c5"
        >
          <strong
            class="c6"
          >
            Nenhum jogo encontrado
          </strong>
          <span
            class="c7"
          >
            Você não adicionou nenhum jogo a sua bolsa de compras.
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="HighlightCarousel"
  />
</div>
`;
